src/bin/main.rs

```diff
@@ -114,27 +126,38 @@ fn main() {
                     Ok(config) => match toml::from_str(&config) {
                         Ok(config) => config,
                         Err(error) => {
-                            eprintln!(
+                            return Err(format!(
                                 "ERROR: Could not parse config file \"{}\": {}",
                                 file_name, error
-                            );
-                            return;
+                            ));
                         }
                     },
                     Err(error) => {
-                        eprintln!(
+                        return Err(format!(
                             "ERROR: Could not read config file \"{}\": {}",
                             file_name, error
-                        );
-                        return;
+                        ));
                     }
                 }
             }
         }
     };
-
     let paths = any_config.paths.unwrap_or_default();
 
+    let root = matches
+        .value_of("root")
+        .map(|s| s.to_string())
+        .or_else(|| paths.root.as_ref().map(|s| s.to_string()))
+        .map_or_else(|| PathBuf::from("."), |r| PathBuf::from(r));
+
+    if let Err(err) = std::env::set_current_dir(&root) {
```

Comment:
I am not so sure this option is pulling its weight if this is its only effect. One can easily change the working directory in the script that calls Yarner. But I guess you arrived at this with a particular use case in mind?

Reply:
I stumbled when trying to set up a small test project combining Yarner and MdBook. Wanting Yarner sources in a sub-directory would require the `Yarner.toml` to be in the sub-directory too (in contrast to `book.toml`), and multiple `cd`-ing to navigate around. Otherwise, doc output would end in a sub-sub-folder (i.e. a sub-folder of `docs`).

Reply:
But then maybe it would be nicer to support having Yarner.toml somewhere else besides the working directory (like Cargo's manifest-path parameter)?

Reply:
But for the structure I intended that would mean to navigate into the sub-directory, and then "call" the toml file from a level above. Hm, this is even more typing and fairly confusing...

Reply:
At least with Cargo, the idea is that all paths in the manifest at relative to the path of the manifest. The manifest-path argument then allows one to start Cargo from anywhere by pointing it at a manifest.

Reply:
> Otherwise, doc output would end in a sub-sub-folder (i.e. a sub-folder of docs).

Why can't the documentation output path include a ".." to end up higher in the folder hierarchy?

Reply:
It can, but that does not solve the problem. Here is an example what happens with the current (i.e. old) implementation. We start with this structure:

```
project
  |--  src/
  |      |--  File-1.md
  |      `--  File-2.md
  `--  Yarner.toml
```

Let's say output folders are `code` and `docs`. We get this:

```
project
  |--  code/
  |--  docs
  |      `--  src
  |             |--  File-1.md
  |             `--  File-2.md
  |--  src/
  |      |--  File-1.md
  |      `--  File-1.md
  `--  Yarner.toml
```

What works is this, with `Yarner.toml` in `src`, running the command from `src` and having output paths `../code` and `../docs`:

```
project
  `--  src/
         |--  File-1.md
         |--  File-1.md
         `--  Yarner.toml
```

Results in this structure:

```
project
  |--  code/
  |--  docs
  |       |--  File-1.md
  |       `--  File-2.md
  `--  src/
         |--  File-1.md
         |--  File-1.md
         `--  Yarner.toml
```

Reply:
> IMO more severe, "hides" the Yarner.toml.

Yes, this is not helpful. I also see now that there is difference to changing the working directory via the shell: The point here is to change it after loading `Yarner.toml`. So all paths are relative to root, except for `Yarner.toml` which is in the folder to which `root` is relative to.

You could consider using the same mechanism as you already employ for `code_files` and `doc_files` also for `files` to e.g. strip the first path component to the keep the number of distinct mechanisms smaller, gain a bit of flexibility and increase the return on investment for learning the path manipulation language.

Reply:
Yes I was already thinking about that option. First, I liked the option for the higher flexibility, but one could also break hyperlinks between doc files when defining different replacement "patterns" for different files/globs. I was not really happy with that, but maybe I should think about it again...

Reply:
As path modifications like for copying code and doc files would also impact (or conflict with) transclusions and link-following, I with stay with the current solution.